test/test_node.go

```diff
@@ -54,42 +64,74 @@ func getGoModuleVersion(pkg string) string {
 	return strings.TrimSpace(string(out))
 }
 
-func getSimdChain() *ChainType {
+func getHeighlinerChain(
+	chain,
+	version,
+	binary,
+	bech32Prefix string,
+	pubKeyAsBech32 bool,
+	preGenTx func(tn *TestNode) error,
+) *ChainType {
 	return &ChainType{
-		Repository: "ghcr.io/strangelove-ventures/heighliner/sim",
-		Version:    getGoModuleVersion("github.com/cosmos/cosmos-sdk"),
-		Bin:        "simd",
-		Ports: map[docker.Port]struct{}{
-			"26656/tcp": {},
-			"26657/tcp": {},
-			"9090/tcp":  {},
-			"1337/tcp":  {},
-			"1234/tcp":  {},
-		},
+		Repository:     fmt.Sprintf("ghcr.io/strangelove-ventures/heighliner/%s", chain),
+		Version:        version,
+		Bin:            binary,
+		Bech32Prefix:   bech32Prefix,
+		PubKeyAsBech32: pubKeyAsBech32,
+		Ports:          cosmosNodePorts,
+		PreGenTx:       preGenTx,
 	}
 }
 
+func getSimdChain() *ChainType {
+	return getHeighlinerChain("sim", getGoModuleVersion("github.com/cosmos/cosmos-sdk"), "simd", "cosmos", false, nil)
+}
+
+func getSentinelChain(ctx context.Context, version string) *ChainType {
+	// sets "approve_by" in the genesis.json
+	// this is required for sentinel, genesis validation fails without it.
+	sentinelGenesisJSONModification := func(tn *TestNode) error {
+		genesisJSON := path.Join(tn.NodeHome(), "config", "genesis.json")
+		address, err := tn.Bech32AddressForKey(valKey)
+		if err != nil {
+			return err
+		}
+		command := []string{"sed", "-i", fmt.Sprintf("s/\"approve_by\": \"\"/\"approve_by\": \"%s\"/g", address), genesisJSON}
```

Comment:
Use a raw string here for readability: `` `s/"approve_by": ...` ``

Or probably more preferable to deserialize the JSON and modify it in Go, since `sed -i` works differently on Linux vs BSD/macOS.

Reply:
the genesis file can be quite large, and we'd have to capture every property in the structs to make sure it would be serialized properly again. This is also just an example of how you could run the tests for a different chain, and the code is not currently used in the test (gated by an if false). I changed it to a raw string for readability for now